Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp

```diff
@@ -760,6 +760,31 @@ Seconds StatsTimestampConverter::convertFromMonotonicTime(Seconds value) const
     return Performance::reduceTimeResolution(newTimestamp.secondsSinceEpoch());
 }
 
+void forEachTransceiver(const GRefPtr<GstElement>& webrtcBin, Function<bool(GRefPtr<GstWebRTCRTPTransceiver>&&)>&& function)
+{
+    GArray* transceivers;
+    g_signal_emit_by_name(webrtcBin.get(), "get-transceivers", &transceivers);
+
+    if (!transceivers || !transceivers->len)
+        return;
+
+    bool shouldCallFunctionAgain = true;
+    for (unsigned index = 0; index < transceivers->len; index++) {
+        WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN; // GLib port
+        auto current = adoptGRef(g_array_index(transceivers, GstWebRTCRTPTransceiver*, index));
+        WTF_ALLOW_UNSAFE_BUFFER_USAGE_END;
+
+        if (!shouldCallFunctionAgain)
+            continue;
+
+        if (function(WTFMove(current)))
+            shouldCallFunctionAgain = false;
+    }
+
+    // Don't free segments because they're moved to local GRefPtrs.
+    g_array_free(transceivers, FALSE);
```

Comment:
From what I understand, you're leaking the array segment here. The g_array contains pointers to the GstWebRTCRTPTransceiver objects so you're taking ownership of here is the memory pointed to that pointer. The array contains pointers. You can and actually have to free the array segment.

Reply:
In the loop above we "adopt" those transceiver refs though, so if we free here wouldn't that be a double free?

Reply:
With the fixup pushed to this PR, the crash no longer absent, and we actually fix an old mem leak as well.